src/lease.rs

```diff
@@ -32,6 +34,23 @@ impl Lease {
         self.local_guard = Some(guard);
         self
     }
+
+    /// Asynchronously releases the underlying lock.
+    pub async fn release(&mut self) {
```

Comment:
I think this should be move the Lease like `async fn release(self)`. This way the Lease, while held, will continue to always mean that you are holding the lock (best effort), so continue to not be possible that the Lease was released then used again somehow.

Reply:
Makes sense.